### PR TITLE
Comments and editorial changes

### PR DESCRIPTION
**Pull request by mariobaldi**

### CODE REVIEW

File: PNA.mdk [hunk 1]
```diff
@@ -144,8 +148,9 @@ division is expected to be:
 
 - main programmable blocks deal solely with individual network
   packets, which are at most one network maximum transmission unit
-  (MTU) in size.
-- message processing is responsible for converting between large
+  (MTU) in size. ** wouldn't this techncially be the size of a network packet?
+  Should we say "Data-link MTU" or "MTU on the wire"? **
```
Comment (jfingerh) [inline PNA.mdk:152]: A glossary item for what we mean by MTU precisely seems warranted here.

File: examples/pna-demo-L2-one-table.p4 [hunk 2]
```diff
@@ -122,9 +119,11 @@ control MainDeparserImpl(
...
     MainParserImpl(),
+    PreControlImpl(),
     MainControlImpl(),
-    MainDeparserImpl(),
-    // The last arg is optional, but leaving it here for the moment
-    MainParserImpl()) main;
+    MainDeparserImpl()
+    // Hoping to make this optional parameter later, but not supported
+    // by p4c yet.
+    //, PreParserImpl()
```
Comment (apinski-cavium) [inline examples/pna-demo-L2-one-table.p4:128]: Below in pna.p4 you delete pre_parser but you keep it in the example here.

File: examples/pna-demo-last-forwarding-action-wins.p4 [hunk 3]
```diff
@@ -163,9 +159,11 @@ control MainDeparserImpl(
...
     MainParserImpl(),
+    PreControlImpl(),
     MainControlImpl(),
-    MainDeparserImpl(),
-    // The last arg is optional, but leaving it here for the moment
-    MainParserImpl()) main;
+    MainDeparserImpl()
+    // Hoping to make this optional parameter later, but not supported
+    // by p4c yet.
+    //, PreParserImpl()
```
Comment (apinski-cavium) [inline examples/pna-demo-last-forwarding-action-wins.p4:168]: Likewise.

File: examples/pna-demo-L2-two-tables.p4 [hunk 4]
```diff
@@ -128,9 +125,11 @@ control MainDeparserImpl(
...
     MainParserImpl(),
+    PreControlImpl(),
     MainControlImpl(),
-    MainDeparserImpl(),
-    // The last arg is optional, but leaving it here for the moment
-    MainParserImpl()) main;
+    MainDeparserImpl()
+    // Hoping to make this optional parameter later, but not supported
+    // by p4c yet.
+    //, PreParserImpl()
```
Comment (apinski-cavium) [inline examples/pna-demo-L2-two-tables.p4:134]: Likewise.

File: examples/pna-example-tcp-connection-state-tracking.p4 [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+}
+
+PNA_NIC(
+    MainParserImpl(),
+    PreControlImpl(),
+    MainControlImpl(),
+    MainDeparserImpl()
+    // Hoping to make this optional parameter later, but not supported
+    // by p4c yet.
+    //, PreParserImpl()
```
Comment (apinski-cavium) [inline examples/pna-example-tcp-connection-state-tracking.p4:299]: Likewise.

File: examples/pna-example-tcp-connection-tracking.p4 [hunk 6]
```diff
@@ -286,9 +354,11 @@ control MainDeparserImpl(
...
 PNA_NIC(
-    PreControlImpl(),
     MainParserImpl(),
+    PreControlImpl(),
     MainControlImpl(),
-    MainDeparserImpl(),
-    // The last arg is optional, but leaving it here for the moment
-    MainParserImpl()) main;
+    MainDeparserImpl()
+    // Hoping to make this optional parameter later, but not supported
```
Comment (apinski-cavium) [inline examples/pna-example-tcp-connection-tracking.p4:361]: Likewise.

File: examples/pna-example-template.p4 [hunk 7]
```diff
@@ -232,9 +178,12 @@ control MainDeparserImpl(
...
 // BEGIN:Package_Instantiation_Example
 PNA_NIC(
-    PreParserImpl(),
-    PreControlImpl(),
     MainParserImpl(),
+    PreControlImpl(),
     MainControlImpl(),
-    MainDeparserImpl()) main;
+    MainDeparserImpl()
+    // Hoping to make this optional parameter later, but not supported
```
Comment (apinski-cavium) [inline examples/pna-example-template.p4:185]: Likewise.

File: pna.p4 [hunk 8]
```diff
@@ -961,15 +909,10 @@ control MainDeparserT<MH, MM>(
...
-    PreControlT<PH, PM> pre_control,
     MainParserT<PM, MH, MM> main_parser,
+    PreControlT<PH, PM> pre_control,
     MainControlT<PM, MH, MM> main_control,
-    MainDeparserT<MH, MM> main_deparser,
-    // pre_parser is optional.  If not specified, it defaults to be the
-    // same as main_parser.  An implementation that has a separate pre
-    // parser is free to optimize away any part of it that is
-    // unnecessary for executing the code in pre_control.
-    PreParserT<PH, PM> pre_parser);
```
Comment (apinski-cavium) [inline pna.p4:972]: You removed it from here.

**Comment by apinski-cavium** [comment]
Or it is the way you did the merge that this shows up.

**Pull request closed without merging**